framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3396,7 +3403,24 @@ NonlinearSystemBase::computeScaling()
     setupScalingGrouping();
 
   auto & field_variables = _vars[0].fieldVariables();
+  std::vector<bool> field_variable_autoscaled(field_variables.size(), true);
+
   auto & scalar_variables = _vars[0].scalars();
+  std::vector<bool> scalar_variable_autoscaled(scalar_variables.size(), true);
+
+  if (_ignore_variables_for_autoscaling.size())
+  {
+    for (MooseIndex(field_variable_autoscaled) i = 0; i < field_variable_autoscaled.size(); ++i)
+      if (find(_ignore_variables_for_autoscaling.begin(),
+               _ignore_variables_for_autoscaling.end(),
+               (*field_variables[i]).name()) != _ignore_variables_for_autoscaling.end())
+        field_variable_autoscaled[i] = false;
+    for (MooseIndex(scalar_variable_autoscaled) i = 0; i < scalar_variable_autoscaled.size(); ++i)
+      if (find(_ignore_variables_for_autoscaling.begin(),
+               _ignore_variables_for_autoscaling.end(),
+               (*scalar_variables[i]).name()) != _ignore_variables_for_autoscaling.end())
+        scalar_variable_autoscaled[i] = false;
+  }
```

Comment:
I would rename `setupScalingGrouping` to something like `setupScalingData` and construct `_field_variable_autoscaled` and `_scalar_variable_autoscaled` data members in there. Then you only do these O(N) algorithms once. Also make sure to use `std::find` as opposed to just `find`, unless you are using some other `find` and then please explicitly apply that scoping operator instead